### PR TITLE
use form body base64 instead of file upload & fix url_path value

### PR DESCRIPTION
**Pull request by goulinkh**
## Description

Fix `upload-assets` with `POST /v1` for assets creation.

## QA

- Start local DBs: `docker-compose up -d`
- You need `python3.7` do be able to start the server locally:
```bash
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt install python3.7 python3-virtualenv python3.7-distutils python3-dotenv-cli

# assets dependencies
apt-get install --no-install-recommends --yes \
python3-setuptools python3-lib2to3 python3-pkg-resources ca-certificates libsodium-dev \
libpng-dev libjpeg-dev libjpeg-progs libmagic1 libmagickwand-dev optipng libpq-dev

virtualenv --python="/usr/bin/python3.7" .venv && source .venv/bin/activate
pip install -r requirements.txt && yarn install
```
- Start the service:
```bash
dotenv yarn start
```
- Generate a token (in another terminal):
```bash
source .venv/bin/activate
dotenv flask create token qa-test
```
- Finally try to upload an asset in a similar matter to this (FYI the POST body is `application/x-www-form-urlencoded`):
https://github.com/canonical/canonicalwebteam.upload-assets/blob/main/canonicalwebteam/upload_assets/uploader.py#L33-L43
- Also make sure that creating an asset through the manager is still working: http://0.0.0.0:8017/manager

### CODE REVIEW

File: webapp/views.py [hunk 1]
```diff
@@ -144,18 +146,15 @@ def create_asset():
     """
     Create a new asset
     """
-    asset = request.files.get("asset")
-    file_content = asset.read()
+    asset = request.values.get("asset")
```
Comment (nottrobin) [inline webapp/views.py:149]: What changed here? This used to be submitted as a file through a standard form and now it's submitted as a base64 string? Why is that?

File: webapp/views.py [hunk 2]
```diff
@@ -144,18 +146,15 @@ def create_asset():
...
     friendly_name = request.values.get("friendly-name")
-    optimize = request.values.get("optimize", False)
-    tags = request.values.get("tags", "")
-    tags = tags.split(",")
-
-    url_path = asset.filename.strip("/")
+    optimize = strtobool(request.values.get("optimize", "false"))
+    tags = request.values.get("tags", "").split(",")
+    url_path = request.values.get("url-path", "").strip("/")
```
Comment (nottrobin) [inline webapp/views.py:153]: What's this for? Doesn't seem to be used

File: webapp/views.py (hunk 1 above)
Comment (goulinkh) [reply webapp/views.py:149]: the standard form case is now part of the manager create view as for the base64 according to [this code](https://github.com/canonical/canonicalwebteam.upload-assets/blob/main/canonicalwebteam/upload_assets/uploader.py#L33-L43) from the upload-assets this is sent as base64.

Comment (nottrobin) [reply webapp/views.py:149]: But I don't think I have the full story here. You say this is to fix upload-assets. Which used to work. So what changed? Is this a change to the old API spec? Or is it to set it back to what the old API spec used to be?

Comment (goulinkh) [reply webapp/views.py:149]: the second one, set it back to the old API spec: https://github.com/canonical/assets.ubuntu.com/blob/legacy/webapp/views.py#L164

**Review by nottrobin — APPROVED** [review]

**Pull request merged**